src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,140 @@
+    void register_nullAge_notOk() {
+        user.setAge(null);
+        assertThrows(UserDataException.class,
+                () -> registrationService.register(user),
+                "Registration user with NULL age"
+                        + " should should throw IncorrectUserDataException.");
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        user.setPassword(null);
+        assertThrows(UserDataException.class,
+                () -> registrationService.register(user),
+                "Registration user with NULL password"
+                        + " should throw IncorrectUserDataException.");
+
+    }
+
+    @Test
+    void register_negativeAge_notOk() {
+        user.setAge(ValidationConstant.NEGATIVE_INVALID_AGE);
+        assertThrows(UserDataException.class,
+                () -> registrationService.register(user),
+                "Registration user with age < 0"
+                        + " should should throw IncorrectUserDataException.");
+    }
+
+    @Test
+    void register_ageGreaterThenMaxAge_notOk() {
+        user.setAge(ValidationConstant.INVALID_LARGE_AGE);
+        assertThrows(UserDataException.class,
+                () -> registrationService.register(user),
+                "Registration user with age > 110"
+                        + " should should throw IncorrectUserDataException.");
+    }
+
+    @Test
+    void register_ageLessThenEighteen_notOk() {
+        user.setAge(ValidationConstant.MIN_INVALID_AGE);
+        assertThrows(UserDataException.class,
+                () -> registrationService.register(user),
+                "Registration user with age < 18"
+                        + " should throw IncorrectUserDataException.");
+    }
+
+    @Test
+    void register_ageValidEighteen_ok() {
+        user.setAge(ValidationConstant.MIN_VALID_AGE);
+        registrationService.register(user);
+        assertTrue(Storage.people.contains(user),
+                "Registration user with age = 18 should add user to Storage.");
+    }
+
+    @Test
+    void register_userExists_notOk() {
+        Storage.people.add(user);
+        assertThrows(UserDataException.class,
+                () -> registrationService.register(user),
+                "Registration user with login that already exists"
+                        + " should throw IncorrectUserDataException.");
+    }
+
+    @Test
+    void register_loginIsEmpty_notOk() {
+        user.setLogin("");
+        assertThrows(UserDataException.class,
+                () -> registrationService.register(user),
+                "Registration user with empty login"
+                        + " should throw IncorrectUserDataException.");
+    }
+
+    @Test
+    void register_passwordLessThenSix_notOk() {
+        user.setPassword(ValidationConstant.INVALID_SHORT_PASSWORD);
+        assertThrows(UserDataException.class,
+                () -> registrationService.register(user),
+                "Registration user with password length less than six"
```

Comment:
it is better to not hardcode values like `six` in the messages. If this value will change you will need to check and fix the all messages in the codebase.